polkadot/xcm/src/lib.rs

```diff
@@ -170,9 +180,13 @@ macro_rules! versioned_type {
 				}
 			}
 		}
-		impl TryFrom<$n> for $v4 {
+		impl<$($($gen),*)?> TryFrom<$n <$($($gen),*)?>> for $v4
+		$(
+			where $($gen: Decode + GetDispatchInfo),*
```

Comment:
this is hard coded only for `VersionedXcm`, since otherwise `TryFrom<v4::Xcm>: v5::Xcm` is not satisfied